File: Yandex/Practicum/Algorightms_and_Data_Structures/3_sprint/E.Покупка_домов/solution.py
import sys


def solution(n: int, k: int, homes: list[int]) -> None:
    """
    Time Complexity O(NlogN)
    Memory Complexity O(1)
    """
    purchased: int = 0
    homes.sort()
    for home in homes:
        if k - home < 0:
            break
        k -= home
        purchased += 1
    sys.stdout.write(str(purchased) + '\n')


if __name__ == '__main__':
    n, k = map(int, sys.stdin.readline().rstrip().split())
    homes: list[int] = list(map(int, sys.stdin.readline().rstrip().split()))
    solution(n, k, homes)

File: Yandex/Practicum/Algorightms_and_Data_Structures/3_sprint/E.Покупка_домов/solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>


void solution(int& n, int& k, std::vector<int>& homes) {
    /*
    Time Complexity O(NlogN)
    Memory Complexity O(1)
    */
    std::sort(homes.begin(), homes.end());
    int purchased = 0;
    for (int& home : homes) {
        if (k - home < 0) {
            break;
        }
        k -= home;
        ++purchased;
    }
    std::cout << purchased << "\n";
}


int main() {
    int n, k;
    std::cin >> n >> k;
    std::vector<int> homes;
    for (int i = 0; i < n; ++i) {
        int home;
        std::cin >> home;
        homes.push_back(home);
    }
    solution(n, k, homes);
}